public/app/features/dashboard/dashgrid/PanelPerformanceMonitor.tsx

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useRef } from 'react';
+
+import { FieldConfigSource } from '@grafana/data';
+import { faro } from '@grafana/faro-web-sdk';
+import appEvents from 'app/core/app_events';
+import { DashboardSavedEvent } from 'app/types/events';
+
+interface Props {
+  children: JSX.Element;
+  isInPanelEdit: boolean;
+  panelType: string;
+  panelId: number;
+  panelTitle: string;
+  panelOptions: any;
+  panelFieldConfig: FieldConfigSource;
+}
+
+export const PanelPerformanceMonitor = (props: Props) => {
+  const startLoadTime = performance.now();
+  const panelIdStr = String(props.panelId);
+
+  const oldPanelOptions = useRef(props.panelOptions);
+  const oldPanelFieldConfig = useRef<FieldConfigSource>(props.panelFieldConfig);
+
+  const savedFieldConfig = useRef<FieldConfigSource>(props.panelFieldConfig);
+  const savedPanelOptions = useRef(props.panelOptions);
+
+  const logChanges = () => {
+    logPanelOptionChanges(savedPanelOptions.current, oldPanelOptions.current);
+    logFieldConfigChanges(savedFieldConfig.current, oldPanelFieldConfig.current);
+
+    //set the old values to the current values for next save
+    oldPanelOptions.current = savedPanelOptions.current;
+    oldPanelFieldConfig.current = savedFieldConfig.current;
+  };
+
+  useEffect(() => {
+    savedPanelOptions.current = props.panelOptions;
+  }, [props.panelOptions]);
+
+  useEffect(() => {
+    savedFieldConfig.current = props.panelFieldConfig;
+  }, [props.panelFieldConfig]);
+
+  const logEvent = (eventName: string, newKey: string, newVal: string, oldVal?: string) => {
```

Comment:
To avoid confusion with the utils declared [here](https://github.com/grafana/grafana/blob/main/packages/grafana-runtime/src/utils/logging.ts), I would consider to rename it to something like `logPanelEvent`.